src/node_process_methods.cc

```diff
@@ -471,6 +473,76 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+#ifdef __POSIX__
+inline int persist_standard_stream(int fd) {
+  int flags = fcntl(fd, F_GETFD, 0);
+
+  if (flags < 0) {
+    return flags;
+  }
+
+  flags &= ~FD_CLOEXEC;
+  return fcntl(fd, F_SETFD, flags);
+}
+
+static void Execve(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kChildProcess, "");
+
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsArray());
+  CHECK(args[2]->IsArray());
+
+  Local<Array> argv_array = args[1].As<Array>();
+  Local<Array> envp_array = args[2].As<Array>();
+
+  // Copy arguments and environment
+  Utf8Value executable(isolate, args[0]);
+  std::vector<std::string> argv_strings(argv_array->Length());
+  std::vector<std::string> envp_strings(envp_array->Length());
+  std::vector<char*> argv(argv_array->Length() + 1);
+  std::vector<char*> envp(envp_array->Length() + 1);
+
+  for (unsigned int i = 0; i < argv_array->Length(); i++) {
+    argv_strings[i] =
+        Utf8Value(isolate, argv_array->Get(context, i).ToLocalChecked())
+            .ToString();
+    argv[i] = argv_strings[i].data();
+  }
+  argv[argv_array->Length()] = nullptr;
+
+  for (unsigned int i = 0; i < envp_array->Length(); i++) {
+    envp_strings[i] =
+        Utf8Value(isolate, envp_array->Get(context, i).ToLocalChecked())
+            .ToString();
+    envp[i] = envp_strings[i].data();
+  }
+  envp[envp_array->Length()] = nullptr;
+
+  // Set stdin, stdout and stderr to be non-close-on-exec
+  // so that the new process will inherit it.
+  for (int i = 0; i < 3; i++) {
+    // Operation failed. Free up memory, then throw.
+    if (persist_standard_stream(i) < 0) {
+      THROW_ERR_PROCESS_EXECVE_FAILED(env, errno);
+    }
+  }
+
+  // Perform the execve operation.
+  // If it returns, it means that the execve operation failed.
+  RunAtExit(env);
+
+  execve(*executable, argv.data(), envp.data());
+  int error_code = errno;
+
+  THROW_ERR_PROCESS_EXECVE_FAILED(env, error_code);
```

Comment:
Given that you're running the `RunAtExit(...)` here, and given that the expectation is for this process to be replaced, I wonder if it would be more appropriate to crash the process here instead of just throwing.

Reply:
Good point.
I update that. Can you check if I used the proper methods to print the errors before aborting or was it there a better way?